### DISCUSSION TITLE
Pulling shared objects in templates / template stacks?

### DISCUSSION THREAD

**Discussion by ie-tlg** [PaloAltoNetworks/pan-os-python#456] [Q&A]
Been trying to wrap my head around templates / templatestacks and locations with pan-python. I cant quite figure out how to access bits, that is in the "Shared" location, if the template default vsys is set? I believe, my question is very similar to https://github.com/PaloAltoNetworks/pan-os-python/issues/451 - if the objects are created manually as "Shared" then a SyslogServerProfile.refreshall(template) returns an empty list, it seems?

**Comment by shinmog** [comment]
Yep, this is similar.

If you want the "shared" scope under a template, then configure your object hierarchy as such:

```python
from panos.panorama import Panorama, Template
from panos.device import Vsys, SyslogServerProfile

pano = Panorama(.....)

tem1 = Template('foo')
pano.add(tem1)

shared = Vsys('shared')
tem1.add(shared)

profileList = SyslogServerProfile.refreshall(shared)
```
